File: android/app/src/main/java/id/my/rizalanggoro/arta/domain/GoldTaxPreference.kt
package id.my.rizalanggoro.arta.domain

import kotlinx.serialization.Serializable

@Serializable
data class GoldTaxPreference(
	val id: Int,
	val userId: Int = 0,
	val carat: Double,
	val taxRate: Double,
	val createdAt: String = "",
	val updatedAt: String = "",
)

File: backend/internal/dto/gold_tax_preference.go
package dto

import "github.com/artafinance/backend/internal/domain"

// GoldTaxPreference represents a user's gold tax preference response DTO.
// deprecated
// @name GoldTaxPreference
type GoldTaxPreference struct {
	ID        uint    `json:"id"`
	UserID    uint    `json:"user_id"`
	Carat     float64 `json:"carat"`
	TaxRate   float64 `json:"tax_rate"`
	CreatedAt string  `json:"created_at"`
	UpdatedAt string  `json:"updated_at"`
}

// @name GoldTaxDTO
type GoldTax struct {
	Data      domain.GoldTaxPreference `json:"data"`
	SellPrice float64                  `json:"sell_price" format:"double"`
}
